filebeat/tests/integration/filestream_truncation_test.go

```diff
@@ -0,0 +1,225 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
@belimawr Did you make sure all these tests are failing without the fix?

Reply:
Yes, I wrote them before start working on the fix. But it has been a while since I ran them, so I'll re-run them without the fix before merging just to be on the safe side.

Reply:
```
--- FAIL: TestFilestreamLiveFileTruncation (8.91s)
    filestream_truncation_test.go:110: expecting offset 500 got 10500 instead

--- FAIL: TestFilestreamOfflineFileTruncation (4.28s)
    filestream_truncation_test.go:150: expecting offset 250 got 750 instead
```